OMCompiler/SimulationRuntime/c/simulation/solver/nonlinearValuesList.c

```diff
@@ -83,62 +83,43 @@ void freeValueList(VALUES_LIST *valueList, unsigned int numberOfList)
 
 void cleanValueList(VALUES_LIST *valueList, LIST_NODE *startNode)
 {
-  LIST_NODE *next, *node;
-  VALUE* elem;
-
-  int i, j;
-  if (startNode == NULL)
+  if(startNode)
   {
-    listClear(valueList->valueList);
-  }
-  else
-  {
-    next = listNextNode(startNode);
-    /* clean list from next node */
-    infoStreamPrint(LOG_NLS_EXTRAPOLATE, 0, "cleanValueList length: %d", listLen(valueList->valueList));
-    updateNodeNext(valueList->valueList, startNode, NULL);
-    removeNodes(valueList->valueList, next);
+    LIST_NODE *next = listNextNode(startNode);
+    if(next)
+    {
+      /* clean list from next node */
+      infoStreamPrint(LOG_NLS_EXTRAPOLATE, 0, "cleanValueList length: %d", listLen(valueList->valueList));
+      updateNodeNext(valueList->valueList, startNode, NULL);
+      removeNodes(valueList->valueList, next);
+    }
   }
+  else listClear(valueList->valueList);
 }
 
 void cleanValueListbyTime(VALUES_LIST *valueList, double time)
```

Comment:
Can you add a documentation? I've read the code and am still not completely sure what the function should do.

Reply:
I'm not entirely sure myself, I just cleaned up the usage of `LIST`s. I suppose the whole `nonlinearValuesList` needs to be refactored, but not in this PR.